packages/bbui/src/Stores/notifications.ts

```diff
@@ -2,9 +2,21 @@ import { writable } from "svelte/store"
 
 const NOTIFICATION_TIMEOUT = 3000
 
+interface Notification {
+  id: string
+  type: string
+  message: string
+  icon: string
+  dismissable: boolean
+  action: (() => void) | null
+  actionMessage: string | null
+  wide: boolean
+  dismissTimeout: number
+}
+
 export const createNotificationStore = () => {
-  const timeoutIds = new Set()
-  const _notifications = writable([], () => {
+  const timeoutIds = new Set<NodeJS.Timeout>()
```

Comment:
We had some issues with this when running on the frontend. 
```suggestion
  const timeoutIds = new Set<ReturnType<typeof setTimeout>>()
```